internal/controller/system/controller.go

```diff
@@ -34,9 +35,14 @@ type Controller interface {
 }
 
 type DefaultController struct {
-	store                      Store
-	listener                   ledgercontroller.Listener
-	parser                     ledgercontroller.NumscriptParser
+	store    Store
```

Comment:
I'm aware that this is hideous:
a. I have to carry around 3 parser version instead of just 1
b. It's error prone, as we are using the interface as type and not the concrete type (e.g. parser `NumscriptInterpreterParser`)

I had to implement it this way because otherwise, if I had to instantiate the runtime in the endpoint (where the "runtime" field is read), It wouldn't apply the other dependencies that were previously applied to the base parser, for example `configuration.NSCacheConfiguration.MaxCount` or the numscript feature flags